src/math/exp_fcn_hifi.c

```diff
@@ -97,6 +118,49 @@ static void mul_s64(ae_int64 in_0, ae_int64 in_1, ae_int64 *ptroutbitshi, ae_int
 	producthi = AE_ADD64(producthihi, carry);
 
 	producthi = AE_ADD64(producthi, product_hl_lh_h);
+#elif EXPONENTIAL_HIFI3 == 1
+
+	ae_int64 producthi_1c;
+	ae_int64 producthi_2c;
+	ae_int64 productlo_2c;
+
+	ae_int64 s0 = AE_SRLI64(in_0, 63);
+	ae_int64 s1 = AE_SRLI64(in_1, 63);
+	bool x_r = (bool)((int)(int64_t)s0 ^ (int)(int64_t)s1);
+
+	ae_int32x2 in0_32 = AE_MOVINT32X2_FROMINT64(AE_ABS64(in_0));
+	ae_int32x2 in1_32 = AE_MOVINT32X2_FROMINT64(AE_ABS64(in_1));
+
+	producthihi = AE_MUL32_HH(in0_32, in1_32);
+	producthilo = AE_MUL32U_LL(in1_32, AE_MOVINT32X2_FROMINT64
+				  ((AE_MOVINT64_FROMINT32X2(in0_32) >> 32)));
```

Comment:
Would AE_SRAI64(x, 32) be faster here? Or would some select function AE_SEL32 do this high word get directly?